sdk/src/client/node_api/plugin/mod.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Custom plugin call
+
+use core::str::FromStr;
+
+use reqwest::Method;
+
+use crate::client::{ClientInner, Result};
+
+impl ClientInner {
+    /// Extension method which provides request methods for plugins.
+    pub async fn call_plugin_route<T>(
+        &self,
+        base_plugin_path: &str,
+        method: &str,
+        method_path: &str,
+        query_params: Vec<String>,
+        request_object: Option<String>,
```

Comment:
Would it make sense to use the presence, or absence, of this field to do a GET or POST? So we don't ask for `method` and deal with the errors that come from it?

Reply:
Ah but I guess you can do a post without a request object?

Reply:
and technically theres other protocols we could support later

Reply:
Yeah just a bit annoying to parse a method name but ok :(